pkg/services/internal/servicequeue/queues_test.go

```diff
@@ -225,47 +246,399 @@ func TestQueue(t *testing.T) {
+			queue := servicequeue.NewQueue2(
+				logging.NewTestLogger(t),
+				"Do3",
+				services...,
+			)
+
+			// when:
+			results, err := queue.All(context.Background(), secondArgument, thirdArgument)
+
+			// then:
+			test.errorExpectation(t, err)
+			assert.Len(t, results, len(test.expectedResults))
+			assert.ElementsMatch(t, test.expectedResults, results)
+		})
+
+		t.Run(name+": Queue3", func(t *testing.T) {
+			// given:
+			services := slices.Map(test.services, func(s TestService) *servicequeue.Service3[string, int, bool, *TestServiceResult] {
+				service := s.NewTest(t)
+				return servicequeue.NewService3(service.Name, service.Do3)
+			})
+
+			// and:
+			queue := servicequeue.NewQueue3(
+				logging.NewTestLogger(t),
+				"Do3",
+				services...,
+			)
+
+			// when:
+			results, err := queue.All(context.Background(), secondArgument, thirdArgument, fourthArgument)
+
+			// then:
+			test.errorExpectation(t, err)
+			assert.Len(t, results, len(test.expectedResults))
+			assert.ElementsMatch(t, test.expectedResults, results)
+		})
+	}
+
+	panicTestCases := map[string]struct {
+		services         []TestService
+		expectedResults  []*servicequeue.NamedResult[*TestServiceResult]
+		errorExpectation func(t assert.TestingT, err error, msgAndArgs ...interface{}) bool
+	}{
+		"handle panic from service": {
+			services: []TestService{
+				TestService{Name: "panicing"}.Panicking(),
+			},
+			expectedResults: []*servicequeue.NamedResult[*TestServiceResult]{
+				// Error is not exactly the same, because it contains more context about the source of the panic, but ErrorIs should be the same.
+				servicequeue.NewNamedResult("panicing", types.FailureResult[*TestServiceResult](errorFromPanic)),
+			},
+			errorExpectation: assert.NoError,
+		},
+		"handle panic of one service between multiple services": {
+			services: []TestService{
+				TestService{Name: "successful"}.Successful(),
+				TestService{Name: "panicing"}.Panicking(),
+				TestService{Name: "ok"}.Successful(),
+			},
+			expectedResults: []*servicequeue.NamedResult[*TestServiceResult]{
+				servicequeue.NewNamedResult("successful", types.SuccessResult(&TestServiceResult{200, "success"})),
+				servicequeue.NewNamedResult("panicing", types.FailureResult[*TestServiceResult](errorFromPanic)),
+				servicequeue.NewNamedResult("ok", types.SuccessResult(&TestServiceResult{200, "success"})),
+			},
+			errorExpectation: assert.NoError,
+		},
+	}
+	for name, test := range panicTestCases {
+		t.Run(name+": Queue", func(t *testing.T) {
+			// given:
+			services := slices.Map(test.services, func(s TestService) *servicequeue.Service[*TestServiceResult] {
+				service := s.NewTest(t)
+				return servicequeue.NewService(service.Name, service.Do)
+			})
+
+			// and:
+			queue := servicequeue.NewQueue(
+				logging.NewTestLogger(t),
+				"Do",
+				services...,
+			)
+
+			// when:
+			results, err := queue.All(context.Background())
+
+			// debug: show example of error from panicing service
```

Comment:
```suggestion
			// debug: show example of error from panicing service
```

Check other occurrences.